extensions/in-memory/negotiation-store-memory/src/main/java/org/eclipse/dataspaceconnector/negotiation/store/memory/InMemoryContractNegotiationStore.java

```diff
@@ -116,6 +123,54 @@ public void delete(String processId) {
         });
     }
 
+    @Override
+    public Stream<ContractNegotiation> queryNegotiations(QuerySpec querySpec) {
+        return readLock(() -> {
+            Stream<ContractNegotiation> negotiationStream = processesById.values().stream();
+            // filter
+            if (CollectionUtil.isNotEmpty(querySpec.getFilterExpression())) {
```

Comment:
This if condition is not needed if the `filterExpression` field is initiated by default with an empty list (that's a good practice overall) because the "empty list" case is already handled by the reduce's identity

Reply:
The `filterExpression` could still be `null`, specifically when `null` is passed in the Builder's `filter()` method. 
`CollectionUtil.isNotEmpty()` checks for null _and_ emptiness.

Reply:
Why someone would set the `filter` as null? A null check could be added on the builder however, like it was done for offset and limit.

Reply:
I'd imagine the `QuerySpec` to be created based on eg. query params of a REST controller, so it seems convenient to just create the `QuerySpec` object directly from the params. 

The `offset` and `limit` are a bit different, because not checking for `null` may cause NPEs during auto-boxing/-unboxing directly in the builder methods.

That said, I don't mind having an `if` statement in the builder method :)